src/pages/tasks/common/helpers.ts

```diff
@@ -24,17 +24,101 @@ export function parseTime(timestamp: number) {
     return;
   }
 
-  return dayjs.unix(timestamp).format('hh:mm:ss');
+  const formattedTimestamp = dayjs.unix(timestamp).format('hh:mm:ss A');
+
+  const [timeValues] = formattedTimestamp.split(' ');
+
+  const [hours, minutes, seconds] = timeValues.split(':');
+
+  if (formattedTimestamp.includes('PM')) {
+    const updatedHours =
+      Number(hours) !== 12 ? Number(hours) + 12 : Number(hours);
+
+    return (
+      updatedHours.toString() +
+      ':' +
+      minutes.toString() +
+      ':' +
+      seconds.toString()
+    );
+  }
+
+  const updatedHours = Number(hours) === 12 ? '00' : hours;
+
+  return (
+    updatedHours.toString() +
+    ':' +
+    minutes.toString() +
+    ':' +
+    seconds.toString()
+  );
 }
 
-export function duration(start: number, stop: number) {
-  const diff = dayjs.unix(stop).diff(dayjs.unix(start), 'seconds');
+export function duration(
+  start: number,
+  stop: number,
+  includedEndDate: boolean
+) {
+  const startDateValue = parseTimeToDate(start);
+  const startTimeValue = parseTime(start);
+
+  const endDateValue = parseTimeToDate(stop);
+  const endTimeValue = parseTime(stop);
+
+  const unixTimestampStart = dayjs(
+    `${startDateValue} ${startTimeValue}`,
+    'YYYY-MM-DD hh:mm:ss'
+  ).unix();
+
+  let unixTimestampEnd = dayjs(
+    `${endDateValue} ${endTimeValue}`,
+    'YYYY-MM-DD hh:mm:ss'
+  ).unix();
+
+  let diff = dayjs
+    .unix(unixTimestampEnd)
+    .diff(dayjs.unix(unixTimestampStart), 'seconds');
+
+  if (!includedEndDate && diff < 0) {
+    unixTimestampEnd = dayjs(
+      `${startDateValue} ${endTimeValue}`,
+      'YYYY-MM-DD hh:mm:ss'
+    ).unix();
+  }
+
+  diff = dayjs
+    .unix(unixTimestampEnd)
+    .diff(dayjs.unix(unixTimestampStart), 'seconds');
 
   if (diff < 0) {
-    return;
+    return '00:00:00';
+  }
+
+  let hours = Math.floor(diff / 3600).toString();
+  diff -= Number(hours) * 3600;
+
+  let minutes = Math.floor(diff / 60).toString();
+  diff -= Number(minutes) * 60;
+
+  let seconds = diff.toString();
+
+  if (Number(hours) < 10) {
+    hours = '0' + hours.toString();
   }
 
-  return new Date(diff * 1000).toISOString().slice(11, 19);
+  if (Number(minutes) < 10) {
+    minutes = '0' + minutes.toString();
+  }
+
+  if (Number(seconds) < 10) {
+    seconds = '0' + seconds.toString();
+  }
+
+  if (stop && startDateValue !== '1970-01-01') {
+    return hours + ':' + minutes + ':' + seconds;
+  } else {
+    return '00:00:00';
+  }
 }
```

Comment:
Same for this.

Reply:
We had a few bugs with duration calculation, especially when we turn on the `end date` and then turn it off in the settings. So, I had to adjust the entire checkout process. I researched the `dayjs` package deeply and found a simpler code solution for this purpose. Now I did the same thing, but with clearer code. Let me know your thoughts.